gio-subclass/tests/simple_application.rs

```diff
@@ -0,0 +1,168 @@
+use std::ops;
+use std::ptr;
+use std::sync::{Once, ONCE_INIT};
+
+extern crate glib_sys as glib_ffi;
+extern crate gobject_sys as gobject_ffi;
+extern crate gio_sys as gio_ffi;
+
+#[macro_use]
+extern crate glib;
+extern crate gio;
+
+use glib::prelude::*;
+use gio::prelude::*;
+use glib::translate::*;
+
+extern crate gobject_subclass;
+extern crate gio_subclass;
+
+use gobject_subclass::object::*;
+
+use gio_subclass::application::*;
+
+
+
+mod imp {
+    use super::*;
+    use std::cell::RefCell;
+
+    pub struct SimpleApplication {
+    }
+
+    static PROPERTIES: [Property; 0] = [];
+
+    impl SimpleApplication {
+        pub fn get_type() -> glib::Type {
+            static ONCE: Once = ONCE_INIT;
+            static mut TYPE: glib::Type = glib::Type::Invalid;
+
+            ONCE.call_once(|| {
+                let t = register_type(SimpleApplicationStatic);
+                unsafe {
+                    TYPE = t;
+                }
+            });
+
+            unsafe { TYPE }
+        }
+
+        fn class_init(klass: &mut ApplicationClass) {
+            klass.install_properties(&PROPERTIES);
+        }
+
+        fn init(obj: &Application) -> Box<ApplicationImpl<Application>> {
+            let imp = Self {
+            };
+            Box::new(imp)
+        }
+    }
+
+    impl ObjectImpl<Application> for SimpleApplication {
+        fn set_property(&self, obj: &glib::Object, id: u32, value: &glib::Value) {
+            let prop = &PROPERTIES[id as usize];
+
+            match *prop {
+                _ => unimplemented!(),
+            }
+        }
+
+        fn get_property(&self, _obj: &glib::Object, id: u32) -> Result<glib::Value, ()> {
+            let prop = &PROPERTIES[id as usize];
+
+            match *prop {
+                _ => unimplemented!(),
+            }
+        }
+    }
+
+    impl ApplicationImpl<Application> for SimpleApplication{
+
+
+        fn local_command_line(&self, application: &Application, arguments: &mut ArgumentList) -> Option<i32> {
+            // application.parent_local_command_line(arguments)
+
+            for line in arguments.iter() {
+                if line.starts_with("--local-"){
+                    println!("handling argument {:?} locally\n", line);
+                }
+            }
+            arguments.remove(0);
+
+            Some(0)
+        }
+
+
+    }
+
+    struct SimpleApplicationStatic;
+
+    impl ImplTypeStatic<Application> for SimpleApplicationStatic {
+        fn get_name(&self) -> &str {
+            "SimpleApplication"
+        }
+
+        fn new(&self, obj: &Application) -> Box<ApplicationImpl<Application>> {
+            SimpleApplication::init(obj)
+        }
+
+        fn class_init(&self, klass: &mut ApplicationClass) {
+            SimpleApplication::class_init(klass);
+        }
+    }
+}
+
+glib_wrapper! {
+    //TODO: is this correct???
+    pub struct SimpleApplication(Object<imp::SimpleApplication>):
+        [Application => InstanceStruct<Application>];
```

Comment:
You also need to add all the parent classes here, i.e. in this case only `gio::Application`. That then gives you `ApplicationExtManual` with its `run` function also